dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Frequency/SignalN.mo

```diff
@@ -0,0 +1,27 @@
+within Dynawo.Electrical.Controls.Frequency;
+
+/*
+* Copyright (c) 2015-2021, RTE (http://www.rte-france.com)
+* See AUTHORS.txt
+* All rights reserved.
+* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, you can obtain one at http://mozilla.org/MPL/2.0/.
+* SPDX-License-Identifier: MPL-2.0
+*
+* This file is part of Dynawo, an hybrid C++/Modelica open source suite of simulation tools for power systems.
+*/
+
+model SignalN "Model for frequency regulation"
+
+import Dynawo.Connectors;
+
+  Connectors.ImPin thetaRef(value(start = 0)) "Voltage angle reference";
+  Connectors.ImPin N "Signal to change the active power reference setpoint of all the generators in the system in p.u (base SnRef)";
+
+equation
+  der(thetaRef.value) = 0;
```

Comment:
not sure I understand: thetaRef and N are not correlated by an equation?

Reply:
Yes, this is the weird thing about this model (same in the cpp model in fact, except that we write theta = 0 instead of der(theta = 0).
In fact we have for each generator i P_i = PRef_i + alpha_i * N. We have i equations but i+1 unknown variables. Writing an additional equation on the voltage angle of one bus in the system (the slack node) solves it.

Reply:
And N is implicitly deduced.